package/firmware/linux-firmware/intel.mk

```diff
@@ -216,6 +216,15 @@ define Package/e100-firmware/install
 endef
 $(eval $(call BuildPackage,e100-firmware))
 
+Package/i915-firmware = $(call Package/firmware-default,Intel I915 firmware,,LICENSE.i915)
+define Package/i915-firmware/install
+	$(INSTALL_DIR) $(1)/lib/firmware/i915
+	$(CP) \
+		$(PKG_BUILD_DIR)/i915/*.bin \
```

Comment:
This is packaging the same files as the `Package/intel-igpu-firmware*` packages. If the existing 3 are missing something please create a new package for it otherwise do not package the same file into to packages.

Reply:
I understand you concern, the total size of all the firmwares is large for a router

```
linux-firmware-20240513/i915$ du -shc *_dmc* | tail -n1
996K    total
linux-firmware-20240513/i915$ du -shc *_guc* | tail -n1
17M     total
linux-firmware-20240513/i915$ du -shc *_huc* | tail -n1
8.8M    total
linux-firmware-20240513/i915$ du -shc * | tail -n1
28M     total
```

But take `amdgpu-firmware` for example, it is larger than `i915-firmware`, but still installed as a whole

```
linux-firmware-20240513$ du -sh amdgpu
83M     amdgpu
```

The reason I think, for the current use cases, if we want to enable GPU, whether for media, graphics or AI, the storage is not a problem for most cases.

It is a balance between simplicity and optimization. 

- Install all the available firmwares as a whole, can support all the platforms, not only the current one but also the future ones. The price to pay is the increased size.
- If we want to minimize the storage, we can customize to install the necessary ones only, even for the target platform only (e.g. ADL) and skip the others. The price to pay is the time to tune.

BTW. i915 is evolving rapidly, as well as the i915 firmware, it is not only for iGPU but also for dGPU now, more function will be added to support new platforms
e.g. `mtl_gsc_1.bin` is for [GSC Firmware](https://github.com/intel/igsc/blob/master/doc/introduction.rst) on MTL

```
linux-firmware-20240513/i915$ find . | grep -v dmc | grep -v guc | grep -v huc
./mtl_gsc_1.bin
```

Reply:
well the main problem is your new `i915-firmware` package will be conflict with `intel-igpu-firmware*`
probably you can create a new package called `intel-igpu-firmware-gsc` and depends on it, or remove `intel-igpu-firmware*` package.

Reply:
Maybe we can consider to make `i915-firmware` as a meta package, which depends on the individual `intel-igpu-firmware-*`.
And can we change the name of `intel-igpu-firmware-*` to `i915-firmware-*`? The prefix `intel-igpu` is misleading, i915 driver/firmware is not for iGPU only, it is for dGPU too

Reply:
> Maybe we can consider to make `i915-firmware` as a meta package, which depends on the individual `intel-igpu-firmware-*`.

Well it would be a little bit hard to create a "real" meta package in OpenWrt. Adding all i915 firmware packages to dependencies is fine IMO.

> And can we change the name of `intel-igpu-firmware-*` to `i915-firmware-*`? The prefix `intel-igpu` is misleading, i915 driver/firmware is not for iGPU only, it is for dGPU too

Sounds good ;)